### PR TITLE
Handle BSD sed when updating package.xml

### PR DESCRIPTION
**Pull request by jcarpent**
Sed on OSX and BSD systems is different from Linux version.
I've adapted the sed regular expression to handle these systems.
@nim65s @jmirabel @wxmerkt Could you check on your side I did not break the behaviour on Linux systems?

### CODE REVIEW

**Comment by wxmerkt** [comment]
I just tried it on Ubuntu and it fails:
```
sed: can't read s|<version>[0-9].[0-9].[0-9]</version>|<version>1.2.3</version>|g: No such file or directory
```

Do we have to add the "+" again as well as @jmirabel pointed out in the other PR?

**Comment by jcarpent** [comment]
In fact, it seems the sed command is failing on Linux systems.
I think the best way would be to split between systems.

**Comment by jmirabel** [comment]
Sereval potential solutions:
- @nim65s is `awk` more portable ?
- I don't think we have to care for what is inside the version tag so no need for extended regex. On my system, `sed -i 's/<version>.*<\/version>/foo/'` suffice.

**Comment by jcarpent** [comment]
I think we are done now. I've added the comment of @jmirabel.
Thanks @wxmerkt and @jmirabel for your help.

File: release.cmake [hunk 1]
```diff
@@ -68,7 +69,7 @@ MACRO(RELEASE_SETUP)
       # Update version in package.xml if it exists
       && if [ -f "package.xml" ]; then
           (echo "Updating package.xml to $$VERSION"
-           && sed -i --expression=\"s!<version>[0-9]\\+.[0-9]\\+.[0-9]\\+</version>!<version>$$VERSION</version>!g\" package.xml
+           && sed -i ${SED_I_ARG} -E \"s|<version>.*</version>|<version>$$VERSION</version>|g\" package.xml
```
Comment (jmirabel) [inline release.cmake:72]: I think `-E` is useless. If you want to have the same for both mac and linux:
```suggestion
           && sed -i.bak \"s|<version>.*</version>|<version>$$VERSION</version>|g\" package.xml
           && rm package.xml.bak
```

Comment (jcarpent) [reply release.cmake]: You're correct. I apply the change.

**Pull request merged**